server/src/main/java/com/cloud/network/router/VpcVirtualNetworkApplianceManagerImpl.java

```diff
@@ -834,4 +837,11 @@ public boolean postStateTransitionEvent(final StateMachine2.Transition<State, Vi
         // once from VpcVirtualNetworkApplianceManagerImpl and once from VirtualNetworkApplianceManagerImpl itself
         return true;
     }
+
+    private Nic getNicWithUpdatedDeviceId(final long nicId, int deviceId) {
+        NicVO nic = _nicDao.findById(nicId);
+        nic.setDeviceId(deviceId);
+        _nicDao.update(nic.getId(), nic);
+        return nic;
+    }
```

Comment:
method name is not clear, Either define & use separate methods (for "updating nic device id" and "getting nic with device id"), or change the name appropriately.

Reply:
`getNicAndUpdtaeDeviceId()` might have been more accurate but this is clearly stating what it does on the tin (to me) , Can you be more explicit about what you want @sureshanaparti ?

Reply:
> `getNicAndUpdtaeDeviceId()` might have been more accurate but this is clearly stating what it does on the tin (to me) , Can you be more explicit about what you want @sureshanaparti ?

simply "_updateNicWithDeviceId()_" returning nic would be ok.

Reply:
@sureshanaparti `updateNicWithDeviceId()` looks good to me. @DaanHoogland what's your opinion ?

Reply:
yes, fine. the only problem i have is with my own spelling; `getNicAndUpdtaeDeviceId()` misses a 'k' ;)

Reply:
@sureshanaparti @DaanHoogland thanks guys. code is changed.